### PR TITLE
App names provided by the User Agent info

### PR DESCRIPTION
**Pull request by m-mcgowan**
# Problem Context

Adds the "app" property to the User Agent info for Hosted and Hostless apps. "Events" means that the app-specific events also include the `app` field.

- [x] NF1 - Hosted Arduino,  Events
- [x] NF4 - Hosted Arduino, Events
- [x] NF5 - Hosted CircuitPython
- [x] NF6 - Hostless
- [x] NF7 - Hostless
- [x] NF8 - Hosted Arduino, Events
- [x] NF9 - Hosted Arduino & Zephyr,  Events
- [x] NF10 - Hosted Arduino (via APP_NAME define.)
- [x] NF11 - Hosted Arduino, Events
- [x] NF12 - Hosted Arduino, Events
- [x] NF13 - Hosted Arduino, Events
- [x] NF14 - Hostless
- [x] NF16 - Hostless
- [x] NF17 - Hostless
- [x] NF18 - Hosted Arduino, Events
- [x] NF19 - Hostless
- [x] NF20 - Hostless
- [x] NF21 - Hostless
- [x] NF22 - Hostless
- [x] NF23 - Hostless
- [x] NF24 - Hostless
- [x] NF25 - Hostless
- [x] NF26 - WiP (in its own PR.)

Not updated

 - NF2 (based on Sparrow Reference firmware.)
 - NF3 ("")
 - NF28 ("")
 - NF29 ("")
 - NF30 ("") 
 - NF31 ("")
 - NF32 ("")

## TODO

 - [ ] How do we want to add the UA info to Sparrow Gateways? Some options:
  - don't bother
  - add instructions in README.md to add `"body":{"app":"nfX"} to the `hub.set` command from the tutorial.
  - expand the reference firmware to allow APP_NAME to be defined, which adds an `app` field to sparrow events.
  - others?

- [x] - write up a note about how to add the app name to the UA info for Hosted Arduino/Python/Zephyr apps, and Hostless apps.

### CODE REVIEW

**Comment by m-mcgowan** [comment]
[PoC for adding the app name to the UA data for hosted and hostless N50 apps.](https://trello.com/c/wvqBEm9W/165-poc-for-adding-the-app-name-to-the-ua-data-for-hosted-and-hostless-n50-apps)

**Review by bsatrom — APPROVED** [review]

**Comment by m-mcgowan** [comment]
Added app name to all Hosted and Hostless. The The two unresolved TODO items aren't blockers to merging.

File: 05-analog-signage/firmware/code.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 i2c = board.I2C()
 card = notecard.OpenI2C(i2c, 0, 0, debug=False)
 
-req = {"req": "hub.set", "mode": "continuous", "sync": True}
+req = {"req": "hub.set", "mode": "continuous", "sync": True, "body": {"app":"nf5"}}
```
Comment (m-mcgowan) [inline 05-analog-signage/firmware/code.py:30]: This is not correct. `note-python` does support the user-agent info, so I need to rework this (and test.)

Comment (m-mcgowan) [reply 05-analog-signage/firmware/code.py]: `note-python` updated with user agent support.

**Review by bsatrom — APPROVED** [review]

**Pull request merged**